src/ucp/core/ucp_worker.c

```diff
@@ -1156,8 +1157,14 @@ ucs_status_t ucp_worker_iface_init(ucp_worker_h worker, ucp_rsc_index_t tl_id,
         }
     }
 
-    context->mem_type_access_tls[context->tl_mds[resource->md_index].
-                                 attr.cap.access_mem_type] |= UCS_BIT(tl_id);
+    access_mem_type = context->tl_mds[resource->md_index].attr.cap.access_mem_type;
+    context->mem_type_access_tls[access_mem_type] |= UCS_BIT(tl_id);
+
+    if (!UCP_MEM_IS_HOST(access_mem_type) &&
+        !(UCS_BIT(access_mem_type) & context->mem_type_detect_bitmap)) {
+        ucs_fatal("%s memoty type support is requested with out it's detection TLS",
```

Comment:
ucs_warn("%s memory type supported but is not detectable")

the fact that we have some UCT which supports memtype does not mean the user is going to pass such pointer. so it doesn't have to be fatal 

also, we are we doing this check when worker is created, and not earlier when the context is created? then will not need to add mem_type_detect_bitmap

Reply:
changed to ucs_warn()
moved the check is context creation time